src/pybind/mgr/orchestrator/_interface.py

```diff
@@ -1282,6 +1293,9 @@ def service_name(self):
         if self.daemon_type in ['mds', 'nfs']:
             _s_name = self.daemon_id.split('.')[0]
             return 'mds.%s' % _s_name
+        if self.daemon_type == 'iscsi':
+            s_name = '.'.join(self.daemon_id.split('.')[0,2])
+            return 'iscsi.%s' % s_name
```

Comment:
This is what I had to add to make it not an orphan daemon. Tacking the instance_id and hostname parts of the daemon_id.